### DISCUSSION TITLE
Get network status

### DISCUSSION THREAD

**Discussion by icyflatline** [doronz88/pymobiledevice3#321] [Q&A] [answered]
Hi,
Is it possible via pymobiledevice3 somehow the get the current network status of the device?
Meaning:
 - Whether device is currently connected to WiFi/CellularData/None/AirplaneMode;
 - Name of the current WiFI SSID;
 - Current device IP address/SubnetMask/DHCP server/DNS;
 
 P.S.
 Is it technically feasible to make an iOS device connect to WiFi network programmatically, via CLI?
 
 Thanks in advance.

**Comment by doronz88** [comment, marked as answer]
I know at least "AirplaneMode" could be queried using the MobileGestalt API:

```shell
pymobiledevice3 diagnostics mg AirplaneMode
```

I'm not sure what other keys are exported via MobileGestalt or IORegistry, which I guess could in theory answer these questions (and are fully accessible through the `diagnostics` sub-command).

Also, extracting more information and making the device perform different API calls if of course possible if the device is jailbroken. This project focuses solely on the APIs Apple allows accessing on non-jailbroken devices.

If you are intereseted on testing stuff on a jailbroken device for automation, maybe this project could interest you more:
https://github.com/doronz88/rpc-project

**Reply by icyflatline** [reply]
Thank you very much!
